### PR TITLE
[recast] Export unofficial target

### LINKED ISSUE #18649: [recast] How to find the library?
The following in my projects CMakeLists.txt fails:
````
find_package(recast REQUIRED)
````
It is not clear which package name to look for. I cannot find this library via vcpkg currently.

Finding other packages from vcpkg works fine in the same CMakeLists.txt
Other libraries provide this information about how to use them after build, it would be very helpful here.

recast build output from vcpkg:
````
(base) PS V:\Coding\vcpkg> .\vcpkg.exe install recast:x64-windows-static --binarysource=clear
Computing installation plan...
The following packages will be built and installed:
    recast[core]:x64-windows-static -> 1.5.1-3
Detecting compiler hash for triplet x64-windows-static...
Starting package 1/1: recast:x64-windows-static
Building package recast[core]:x64-windows-static...
-- Using V:/Coding/vcpkg/downloads/recastnavigation-recastnavigation-1.5.1.tar.gz
-- Cleaning sources at V:/Coding/vcpkg/buildtrees/recast/src/1.5.1-6d6c5e6c37.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source V:/Coding/vcpkg/downloads/recastnavigation-recastnavigation-1.5.1.tar.gz
-- Using source at V:/Coding/vcpkg/buildtrees/recast/src/1.5.1-6d6c5e6c37.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows-static
-- Building x64-windows-static-dbg
-- Building x64-windows-static-rel
-- Installing: V:/Coding/vcpkg/packages/recast_x64-windows-static/share/recast/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package recast[core]:x64-windows-static... done
Installing package recast[core]:x64-windows-static...
Installing package recast[core]:x64-windows-static... done
Elapsed time for package recast:x64-windows-static: 8.154 s

Total elapsed time: 12.27 s
````

### PR DESCRIPTION
**Pull request by JackBoosY**
Export the target that the user can use recast simply:
```
The package recast:x86-windows provides CMake targets:

    find_package(unofficial-recast CONFIG REQUIRED)
    target_link_libraries(main PRIVATE recast)
```

Fixes #18649.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@Maxomann  Can you please test my changes?

Thanks.

**Comment by Maxomann** [comment]
@JackBoosY Thanks for your changes!
I have tried to set up the RecastDemo project from the recastnavigation/recastnavigation repository. So far I have noticed two things:

First:
`#include` directives (for example to: `#include <recast/Recast/Recast.h>`) only work if there is at least one other vcpkg library linked to the RecastDemo target.
So this:
`target_link_libraries(RecastDemo PRIVATE recast [...])` would result in `fatal error C1083: Cannot open include file: 'recast/Recast/Recast.h': No such file or directory`

While this (linking completely unrelated vcpkg library `nlohmann_json::nlohmann_json` ):
`target_link_libraries(RecastDemo PRIVATE recast nlohmann_json::nlohmann_json  [...])` does not produce said error.

----

Second:
Upon further investigation it seem that certain `#include` directives in the library itself don't work correctly.
This leads to build errors like: `[...]\vcpkg\installed\x64-windows\include\recast\DetourCrowd\DetourPathCorridor.h(22): fatal error C1083: Cannot open include file: 'DetourNavMeshQuery.h': No such file or directory`

`#include "DetourNavMeshQuery.h"` is expected to be in the same folder as `DetourPathCorridor.h`, but they are both in different sub-folders. Namely:

- `[...]\vcpkg\installed\x64-windows\include\recast\DetourCrowd` and
- `[...]\vcpkg\installed\x64-windows\include\recast\Detour`

----

I hope this was helpful. If there's anything I can help with further, feel free to let me know.

**Comment by JackBoosY** [comment]
@Maxomann Please try again.

Thanks.

File: versions/r-/recast.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "949b280dbfefbc4ad6c1af2141e2632786baa30d",
```
Comment (JackBoosY) [inline versions/r-/recast.json:4]:
```suggestion
      "git-tree": "2f34af0b398b4c37f0cdf3fcd8dc9db965e4dd5e",
```

**Comment by Maxomann** [comment]
These changes allowed me to build the demo project: https://github.com/Maxomann/vcpkg/commit/450c7afbc21097e6b71ceb3fe0e6c32f5c4ad69d

**Review by PhoebeHui — APPROVED** [review]

**Comment by PhoebeHui** [comment]
Cmake doesn't provide findmodule.cmake, so this LGTM.

**Pull request merged**